helios/pipeViewer/transactiondb/src/Reader.hpp

```diff
@@ -105,1158 +104,1134 @@ namespace sparta{
+                                pair_t::IntT tmp = 0;
+                                record_file_.read(&tmp, item_size);
+                                pairt.valueVector.emplace_back(std::make_pair(tmp, true));
+
+                                // Finally for a certain field "i", we check if there is a string
+                                // representation for the integral value, by checking,
+                                // if a key with current pair id, current field id, current integral value
+                                // exists in the In-memory String Map. If yes, we grab the value
+                                // and place it in the enum vector.
+                                if(const auto it = stringMap_.find(std::make_tuple(pairt.valueVector[0].first,
+                                                                                   i,
+                                                                                   pairt.valueVector[i].first));
+                                   it != stringMap_.end()) {
+                                    pairt.stringVector.emplace_back(it->second);
+                                    pairt.valueVector[i].second = false;
+                                }
+
+                                // Else, we convert integer value to string
+                                else {
+                                    const auto int_value = pairt.valueVector[i].first;
+
+                                    if (int_value == std::numeric_limits<pair_t::IntT>::max()) {
+                                        // Max value, so probably bad...push empty string
+                                        pairt.stringVector.emplace_back("");
+                                    } else {
+                                        const auto& format_str = pairt.delimVector[i];
+
+                                        std::ios_base::fmtflags format_flags = std::ios::dec;
+                                        std::string_view fmt_prefix = "";
+
+                                        if(format_str == PairFormatter::HEX) {
+                                            format_flags = std::ios::hex;
+                                            fmt_prefix = "0x";
+                                        }
+                                        else if(format_str == PairFormatter::OCTAL) {
+                                            format_flags = std::ios::oct;
+                                            fmt_prefix = "0";
+                                        }
+
+                                        std::ostringstream int_str;
+                                        int_str << fmt_prefix;
+                                        int_str.setf(format_flags, std::ios::basefield);
+                                        int_str << int_value;
+                                        pairt.stringVector.emplace_back(int_str.str());
```

Comment:
Format flags get applied here before being passed on to Argos. Note that the Python side of Argos didn't need any changes for format flags to work since everything gets passed to it as a string.